CONTRIBUTING.md

```diff
@@ -10,12 +10,33 @@ It is intended to be followed without needing support (i.e. should explain thing
 
 ## How we teach
 
+### Learning by doing
+
 We primarily teach via exercises and projects. We describe outcomes, show examples, and give references (either terms to Google, or links to useful material).
 
+### Don't reinvent the wheel
+
 Where we think there isn't a useful / accurate / well-targeted existing resource, we may write explanations ourselves, but this should be rare - many excellent people have come before us!
 
+### Make it challenging
+
 We don't explain every step along the way - working out the steps is an important part of learning. Deciding exactly how many steps to explain is a tricky balance.
 
+### Fundamentals first
+
+As far as possible and reasonable, we teach foundational topics _before_ teaching abstractions. The goal is that learners can re-apply their knowledge to new situations by re-combining the fundamentals, rather than simply being able to use a specific tool.
```

Comment:
By abstractions do we really mean "frameworks"? I'm not quite sure how to read it...

Also, this backs up my thoughts in https://github.com/CodeYourFuture/immersive-go-course/issues/33 that we should do a little "don't use Cobra yet" project to focus on some fundamentals :)

Reply:
Yes, but also things like AWS Lambda before running on VMs.